packages/rxjs-errors/readme.md

```diff
@@ -0,0 +1,144 @@
+# RxJS error utils
+
+Utils that help work with errors in rxjs observable's
+
+## Classes
+
+- `RxjsFilterError` - custom error used by error operators.
+
+## Available operators
+
+- [`filterWithError`](#filterwitherror) - works like `filter` from rxjs, but throw an error instead of ignoring the value.
+- [`rethrowExceptionWithLogger`](#rethrowexceptionwithlogger) - will catch any error in the observable, create a
+  new `RxJSFilterError` (or any custom) with custom message and log it
+  using [@rnw-community/nestjs-rxjs-logger](../nestjs-rxjs-logger). If it catches `RxJSFilterError` (or any custom), it will
+  just rethrow that exception as is.
+
+## Usage examples
+
+### filterWithError
+
+If the condition does not pass, will throw `RxJSFilterError` exception with passed in message.
+
+```typescript
+import { of } from 'rxjs';
+import { filterWithError } from '@rnw-community/rxjs-errors';
+
+of('C')
+    .pipe(filterWithError(letter => letter === 'A', letter => `Wanted A, but got ${letter}`))
+    .subscribe(value => console.log(value));
+```
+
+Passing custom error class:
+
+```typescript
+import { filterWithError } from '@rnw-community/rxjs-errors';
+import { BadRequestException } from '@nestjs/common';
+import { of } from 'rxjs';
+
+of('F')
+    .pipe(filterWithError(letter => letter === 'A', letter => `Wanted A, but got ${letter}`, BadRequestException))
+    .subscribe(value => console.log(value));
+```
+
+Passing a type guard:
+
+```typescript
+import { filterWithError } from '@rnw-community/rxjs-errors';
+import { of } from 'rxjs';
+
+const value: unknown = '';
+
+const isString = (val: unknown): val is string => typeof val === 'string';
+
+of(value) // `unknown` here
+    .pipe(filterWithError(isString, value => `Wanted string, but got ${typeof value}`))
+    .subscribe(value /* `string` here */ => {
+        console.log(value);
+    });
+```
+
+You could also just pass in a message instead of a function:
+
+```typescript
+import { of } from 'rxjs';
+import { filterWithError } from '@rnw-community/rxjs-errors';
+
+of('C')
+    .pipe(filterWithError(letter => letter === 'A', 'Invalid letter recevied'))
+    .subscribe(value => console.log(value));
+```
+
+### rethrowExceptionWithLogger
```

Comment:
lets change the naming, this operators are connected to each other - so they should follow the same naming pattern, like
`filterWithException` -> `rethrowExceptionWithLogger`

Reply:
And do we really need `WithLogger` in the naming? an we loose logger type from our package `NestJSRxJSLoggerService` just to something that can log `error` and is more generic? and provide a fallback to send message to `console.error`

Reply:
The idea is to give ability to use this operators on the FE, thus we should remove all BE deps